docs/weave-tutorial/03-codec.md

```diff
@@ -0,0 +1,219 @@
+---
+id: codec
+title: Codec
+sidebar_label: Codec
+---
+
+> [PR#1](https://github.com/iov-one/tutorial/pull/1): _Create order book models_
+
+Weave defines `Marshaller` and `Persistent` interface standards. These interfaces are automatically implemented by any code autogenerated by protoc from protobuf files, and we recommend using .proto files to specify the serialization format for any persistent data in our application (internal state as well as transactions and messages). However, if you have never worked with protobuf, this might be a bit of a challenge, so we explain a workflow that we use in weave based projects.
```

Comment:
```suggestion
Weave defines `Marshaller` and `Persistent` interface standards. These interfaces are automatically implemented by any code autogenerated by protoc from protobuf files, and we recommend using .proto files to specify the serialization format for any persistent data in our application (internal state as well as transactions and messages). However, if you have never worked with protobuf, this might be a bit of a challenge, so here we explain a workflow that we use in weave- based projects.
```